### PR TITLE
Feature/bubble chart

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

**Comment by tnagorra** [comment]
I get this error on COVID Dashboard
image

**Comment by AdityaKhatri** [comment]
Cannot replicate the issue.

File: src/components/IndicatorMap/index.tsx [hunk 1]
```diff
@@ -134,6 +206,76 @@ function IndicatorMap(props: Props) {
                     </MapTooltip>
                 )}
             </MapSource>
+            <MapSource
+                sourceKey="neal-centroids"
```
Comment (tnagorra) [inline src/components/IndicatorMap/index.tsx]: I think you mean `nepal`

File: src/utils/common.ts [hunk 2]
```diff
@@ -144,3 +144,64 @@ export const generateChoroplethMapPaintAndLegend = (
...
     return { min: minValue, paint: emptyPaint, legend: emptyLegend };
 };
+
+export const generateBubbleMapPaintAndLegend = (
+    minVal: number = 0,
+    maxValue: number = 0,
+    maxRadius: number = 50,
+) => {
+    let minValue = minVal;
+    if (minValue === maxValue) {
```
Comment (tnagorra) [inline src/utils/common.ts:161]: Can we also add this to choropleth?

File: src/components/BubbleLegend/styles.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        align-items: center;
+        flex-shrink: 0;
+        padding-right: var(--spacing-medium);
+        font-size: var(--font-size-small);
+
+        &:before {
+            opacity: 0.6;
+            margin-right: var(--spacing-small);
+            border-radius: 50%;
+            background-color: #de2d26;
```
Comment (tnagorra) [inline src/components/BubbleLegend/styles.css]: Can we let user pass this value?

File: src/components/BubbleLegend/styles.css [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        display: flex;
+        align-items: center;
+        flex-shrink: 0;
+        font-size: var(--font-size-small);
+
+        &:before {
+            opacity: 0.6;
+            margin-right: var(--spacing-small);
+            border-radius: 50%;
+            background-color: #01665e;
```
Comment (tnagorra) [inline src/components/BubbleLegend/styles.css]: Can we let user pass this value?

File: src/views/Covid19/index.tsx [hunk 5]
```diff
@@ -133,6 +135,7 @@ function Covid19(props: Props) {
     const [indicatorListPending, indicatorListResponse] = useRequest<MultiResponse<Indicator>>(
         indicatorListGetUrl,
     );
+    const indicatorList = indicatorListResponse?.results || emptyIndicatorList;
```
Comment (tnagorra) [inline src/views/Covid19/index.tsx]: You don't need to assign empty list as default.
You can simply do `indicatorList?.find(i => indicatorKeySelector(i) === selectedIndicator)`

File: src/views/Covid19/index.tsx [hunk 6]
```diff
@@ -144,27 +147,80 @@ function Covid19(props: Props) {
...
-        [mapState],
+        [choroplethMapState],
     );
 
+    const {
+        mapPaint: bubblePaint,
+        legend: bubbleLegend,
+    } = useMemo(() => {
+        const valueList = bubbleMapState
+            .filter(d => isDefined(d.value)).map(d => Math.abs(d.value));
```
Comment (tnagorra) [inline src/views/Covid19/index.tsx]: ``
```suggestion
        const valueList = bubbleMapState
            .map(d => d.value)
            .filter(isDefined)
            .map(Math.abs);
```

File: src/views/Dashboard/index.tsx [hunk 7]
```diff
@@ -68,59 +74,41 @@ const fiveWOptions: FiveWOption[] = [
...
-        setAttribute,
-    ] = React.useState<'indicator' | 'fiveW'>('fiveW');
+    const [invertMapStyle, setInvertMapStyle] = useState(false);
 
     const indicatorListGetUrl = `${apiEndPoint}/core/indicator-list/`;
     const [
         indicatorListPending,
         indicatorListResponse,
     ] = useRequest<MultiResponse<Indicator>>(indicatorListGetUrl);
+    const indicatorList = indicatorListResponse?.results || emptyIndicatorList;
```
Comment (tnagorra) [inline src/views/Dashboard/index.tsx]: No need to set default value for list

File: src/views/Dashboard/index.tsx [hunk 8]
```diff
@@ -133,27 +121,77 @@ const Dashboard = (props: Props) => {
...
+        [choroplethMapState],
     );
 
     // const pending = mapStatePending || indicatorListPending;
+    const {
+        mapPaint: bubblePaint,
+        legend: bubbleLegend,
+    } = useMemo(() => {
+        const valueList = bubbleMapState
+            .filter(d => isDefined(d.value)).map(d => Math.abs(d.value));
```
Comment (tnagorra) [inline src/views/Dashboard/index.tsx]: Same as suggested above.

**Pull request merged**